Userland/Libraries/LibUnicode/Normalize.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022, mat <???>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <AK/Utf32View.h>
+#include <AK/Vector.h>
+
+namespace Unicode {
+
+[[nodiscard]] Vector<u32> normalize_nfd(Utf32View string);
+
+[[nodiscard]] Vector<u32> normalize_nfc(Utf32View string);
+
+[[nodiscard]] Vector<u32> normalize_nfkd(Utf32View string);
+
+[[nodiscard]] Vector<u32> normalize_nfkc(Utf32View string);
+
+enum class NormalizationForm {
+    NFD,
+    NFC,
+    NFKD,
+    NFKC
+};
+
+[[nodiscard]] Vector<u32> normalize(NormalizationForm form, Utf32View string);
```

Comment:
UTF-32 makes it easier to just iterate through the code points, which is what most of the algorithms do. Would having a StringView overload that then converts it to UTF-32 be fine?

Reply:
Nevermind, just a public StringView overload is fine.

Reply:
:+1: In case you haven't found it, you can take a `StringView` as a parameter, and wrap it in a `Utf8View` inside your function. That will let you iterate over the same way you're doing now, one decoded code point at a time.